drivers/sdhc/sdhc_spi.c

```diff
@@ -224,7 +224,7 @@ static int sdhc_spi_response_get(const struct device *dev, struct sdhc_command *
 		 */
 		response = dev_data->scratch;
 		end = response + 1;
-		for (i = 0; i < 16; i++) {
+		while (cmd->timeout_ms > 0) {
```

Comment:
this structure is ultimately provided by driver API and I don't think it is good practice to modify it, the contract of the API in sdhc.h does not say it will do that, so we should either update the API contract to mention this or create a stack variable

Reply:
Fixed by adding a stack variable